lib/load_routes.go

```diff
@@ -110,6 +113,47 @@ func loadRoutesFromCS(pool PgxIface, mux *triemux.Mux, backends map[string]http.
 	return nil
 }
 
+func (rt *Router) listenForContentStoreUpdates(ctx context.Context) error {
+	listener := &pgxlisten.Listener{
+		Connect: func(ctx context.Context) (*pgx.Conn, error) {
+			c, err := rt.pool.Acquire(ctx)
+			if err != nil {
+				return nil, err
+			}
+			return c.Conn(), nil
+		},
+	}
+
+	listener.Handle("route_changes", pgxlisten.HandlerFunc(
+		func(ctx context.Context, notification *pgconn.Notification, conn *pgx.Conn) error {
+			rt.CsReloadChan <- true
+			return nil
+		}),
+	)
+
+	err := listener.Listen(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (rt *Router) waitForReload() {
+	for range rt.CsReloadChan {
+		func() {
```

Comment:
I think this can be removed as it duplicates functionality already in `reloadCsRoutes`